PayloadPkg/OsLoader/Rtcm.c

```diff
@@ -0,0 +1,83 @@
+/** @file
+
+  Copyright (c) 2019 - 2021, Intel Corporation. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Library/DebugLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/BootloaderCommonLib.h>
+#include "OsLoader.h"
+
+#define SIGNATURE_IPFW    SIGNATURE_32 ('I', 'P', 'F', 'W')
+#define SIGNATURE_RTCM    SIGNATURE_32 ('T', 'C', 'C', 'R')
+
+typedef VOID (*RTCM_ENTRY)(VOID *Params);
+
+/**
+  Load and Run RTCM.
+
+  This function will load and run RTCM.
+
+  @retval  EFI_NOT_FOUND    RTCM binary file was not found.
+  @retval  EFI_LOAD_ERROR   RTCM binary file loading failed
+  @retval  EFI_SUCCESS      RTCM load and run successfully.
+**/
+EFI_STATUS
+LoadAndRunRtcm(

```

Comment:
I think this function should be renamed to a generic function LoadPeImage(), and it can be put into BootloaderCommonPkg.    LoadComponent() and call the actual entry point can be left out in OsLoader.
Also do you think we can use the similar boot option mechanism to handle it ?  The only difference is this boot option will return ?

Reply:
yes, we could use boot option mechanism to handle it. we need load this kind of image using extra image slots, and we need extend boot option flags and boot image info to indicates this kind of info. let me explore it.